src/index.ts

```diff
@@ -401,61 +466,61 @@ export default class Surreal extends Emitter {
 		}
 	}
 
-	#result(res, resolve, reject) {
+	#result<T>(res: Result<T>, resolve: (value: T) => void, reject: (reason?: any) => void): void {
 		if (res.error) {
 			return reject( new Error(res.error.message) );
 		} else if (res.result) {
 			return resolve(res.result);
 		}
-		return resolve();
+		return resolve(undefined as T);
```

Comment:
```ts
#result<T>(res: Result<T>, resolve: (value: T | undefined) => void, reject: (reason?: any) => void): void {
  if (res.error) {
	  return reject( new Error(res.error.message) );
  } else if (res.result) {
	  return resolve(res.result);
  }
  return resolve(undefined);
}
```
(Sorry, github didn't let me do a code suggestion, see the changes of the resolve type)

I'm not sure if `undefined as T` is even allowed in typescript?
Same change should be done for `#output`.

Reply:
function parameter should be changed to `resolve: (value?: T) => void`.

Reply:
`resolve: (value?: T) => void` had issues where ts didn't like that it could accept `value: undefined | T` instead of just `T`.

In the future this could be easily switched to using async/await which would completely remove the need for this type.